### PR TITLE
Create test for publishing messages

### PR DESCRIPTION
**Pull request by Hannarst**
Using method in `TestingLayer`

### CODE REVIEW

**Comment by Hannarst** [comment]
@hodgestar

File: vxyowsup/tests/test_whatsapp.py [hunk 1]
```diff
@@ -26,8 +26,12 @@ def getDummyCoreLayers():
 def TUMessage_to_PTNode(message):
     # message is TransportUserMessage
     # returns ProtocolTreeNode
-    return TextMessageProtocolEntity(message['content'], to=message['to_addr']
-                                     + '@s.whatsapp.net').toProtocolTreeNode()
+    if message['to_addr']:
+        return TextMessageProtocolEntity(message['content'], to=message['to_addr']
+                                         + '@s.whatsapp.net').toProtocolTreeNode()
+    elif message['from_addr']:
```
Comment (hodgestar) [inline vxyowsup/tests/test_whatsapp.py]: I think it should usually be the case that both `to_addr` and `from_addr` are set, so perhaps we should structure this as:
```python
to_addr = add_whatsapp_suffix(message['to_addr'])
from_addr = add_whatsapp_suffix(message['from_addr'])
return TextMessageProtocolEntity(message['content'], to=to_addr, _from=from_addr)
```
where
```python
def add_whatsapp_suffix(addr):
    if addr:
        addr = addr + '@s.whatsapp.net'
    return addr
```
Thoughts?

File: vxyowsup/tests/test_whatsapp.py [hunk 2]
```diff
@@ -56,6 +60,12 @@ def test_outbound(self):
         node_received = yield self.testing_layer.data_received.get()
         self.assert_nodes_equal(TUMessage_to_PTNode(message_sent), node_received)
 
+    @inlineCallbacks
+    def test_publish(self):
+        message_sent = yield self.testing_layer.send_to_transport(text='Hi Vumi! :)', from_address='meeeeeeeee@s.whatsapp.net')
```
Comment (hodgestar) [inline vxyowsup/tests/test_whatsapp.py]: Woot on the `meeeeeeeee'. :)

**Comment by hodgestar** [comment]
Left one suggestion, otherwise looks good.

File: vxyowsup/tests/test_whatsapp.py (hunk 1 above)
Comment (Hannarst) [inline vxyowsup/tests/test_whatsapp.py]: I think that's explicitly not the case with `TextMessageProtocolEntity` for some reason.

**Comment by Hannarst** [comment]
@hodgestar Ready for review

File: vxyowsup/whatsapp.py [hunk 3]
```diff
@@ -134,19 +130,18 @@ def send_to_human(self, text, to_address):
 
     @ProtocolEntityCallback("message")
     def onMessage(self, messageProtocolEntity):
-        from_address = messageProtocolEntity.getFrom()
+        from_address = messageProtocolEntity.getFrom(False)
```
Comment (hodgestar) [inline vxyowsup/whatsapp.py]: Woot on discovering the `False` flag.

**Comment by hodgestar** [comment]
:+1:

**Pull request closed without merging**